app/frontend/src/features/auth/login/LoginForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { loginPasswordRoute, resetPasswordRoute } from "../../../routes";
+import { service } from "../../../service";
+import { useIsMounted, useSafeState } from "../../../utils/hooks";
+import { useAuthContext } from "../AuthProvider";
+import useAuthStyles from "../useAuthStyles";
+
+const useStyles = makeStyles((theme) => ({
+  loginOptions: {
+    display: "flex",
+    marginTop: theme.spacing(2),
+  },
+  forgotPasswordLink: {
+    color: "#333333",
+    lineHeight: theme.spacing(5),
+  },
+}));
+
+export default function UsernameForm() {
+  const classes = useStyles();
+  const authClasses = useAuthStyles();
+  const { authState, authActions } = useAuthContext();
+  const authLoading = authState.loading;
+  const [sent, setSent] = useState(false);
+  const [loading, setLoading] = useSafeState(useIsMounted(), false);
+  const [loginWithLink, setLoginWithLink] = useState(true);
+
+  const { handleSubmit, register } = useForm<{ username: string }>();
+  const history = useHistory();
+
+  const onSubmit = handleSubmit(
+    async (data: { username: string; password: string }) => {
+      setLoading(true);
+      authActions.clearError();
+      try {
+        const next = await service.auth.checkUsername(data.username);
+        switch (next) {
+          case LoginRes.LoginStep.INVALID_USER:
+            authActions.authError("Couldn't find that user.");
+            break;
+
+          case LoginRes.LoginStep.NEED_PASSWORD:
+            history.push(loginPasswordRoute, data);
+            //return here to avoid setLoading when
+            //this component is no longer mounted
+            setLoginWithLink(false);
+            return;
+
+          case LoginRes.LoginStep.SENT_LOGIN_EMAIL:
+            setSent(true);
+            break;
+        }
+
+        if (!loginWithLink) {
+          authActions.passwordLogin({
+            username: data.username,
+            password: data.password,
+          });
+        }
+      } catch (e) {
+        authActions.authError(e.message);
+      }
+      setLoading(false);
+    }
+  );
+
+  return (
+    <>
+      {sent && <TextBody>Check your email for a link to log in! :)</TextBody>}
+      <form className={authClasses.form} onSubmit={onSubmit}>
+        <InputLabel className={authClasses.formLabel} htmlFor="username">
+          Email/Username
+        </InputLabel>
+        <TextField
+          className={authClasses.formField}
+          disabled={sent}
+          fullWidth
+          id="username"
+          inputRef={register({ required: true })}
+          name="username"
+          variant="standard"
+        />
+        {!loginWithLink && (
+          <>
+            <InputLabel className={authClasses.formLabel} htmlFor="password">
+              Password
+            </InputLabel>
+            <TextField
+              className={authClasses.formField}
+              fullWidth
+              id="password"
+              label="Enter your password"
```

Comment:
I think make a blank label, since the material ui label floats up and looks weird. tbh we probably need to rethink the textfield.

Reply:
Yes, forgot to remove it there